### PR TITLE
Update Gradle 5

### PR DESCRIPTION
**Pull request by sfriberg**
Allow using Gradle properties as well as ENV variables

### CODE REVIEW

**Comment by sfriberg** [comment]
@thegreystone Do we really care about 9 or move it to 11 compat?

To make it easier in the IDE I wonder if the code should be in two subprojects?

**Comment by thegreystone** [comment]
It was in JDK 9 that the API changes we need to cater for were implemented. That said, I can't imagine anyone running JDK 9 or 10, so it probably wouldn't make much of a difference. Then again, it doesn't hurt to support them. 

That said, I don't mind if testing only happens on one pre-JDK 9 (Oracle JDK) platform and OpenJDK 11.

File: build.gradle [hunk 1]
```diff
@@ -54,8 +52,8 @@ compileJava {
 }
 
 compileJava9Java {
+	options.compilerArgs.addAll(['--release', '11'])
 	sourceCompatibility = 9
-	targetCompatibility = 9
```
Comment (thegreystone) [inline build.gradle]: Not that I mind terribly, but why not leave it to be 9 compatible? Is there anything in particular that you worry about?

**Comment by thegreystone** [comment]
> To make it easier in the IDE I wonder if the code should be in two subprojects?

It's so tiny amounts of code though (especially in the opentracing-0.32.0 branch). Personally I don't mind leaving it as it is. At some point the world will all be on 11 (let's pray to his Noodliness that this will happen before retirement) and we can just move a few classes and drop JDK 7 and 8 compatibility. :)

File: build.gradle (hunk 1 above)
Comment (sfriberg) [reply build.gradle]: Doesn't compile with --release 9 with Java 11. Guessing it wasn't officially support in 9 so it is excluded.

Comment (sfriberg) [reply build.gradle]: Moved it completely to Java11. Let me know if you want me to revert it back to support 9, but it should not be a used JDK version today.

Comment (thegreystone) [reply build.gradle]: That's fine then. It's not supported to run anything on JDK 9 and 10 in production today anyways.

**Pull request merged**